tests/integration/pilot/traffic_shifting_test.go

```diff
@@ -66,57 +68,103 @@ type VirtualServiceConfig struct {
 	Weight2   int32
 }
 
+// Test case for regular echo instances
 func TestTrafficShifting(t *testing.T) {
+	trafficShifting(t, []string{}) // pass empty list to indicate no VM
+}
+
+// Test case for VM traffic shifting. Client a will sent requests to 3 VMs
+// VM image version is default to ubuntu:bionic
+// All other images will be tested in post-submit to reduce runtime
+func TestVMTrafficShifting(t *testing.T) {
+	trafficShifting(t, []string{DefaultVMImage})
+}
+
+// Test case for post-submit to test all OS types/versions
+func TestVMTrafficShiftingPost(t *testing.T) {
+	trafficShifting(t, GetSupportedOSVersion(), label.Postsubmit) // mark it to be post-submit
+}
+
+func trafficShifting(t *testing.T, vmImages []string, label ...label.Instance) {
 	// Traffic distribution
 	weights := map[string][]int32{
 		"20-80":    {20, 80},
 		"50-50":    {50, 50},
 		"33-33-34": {33, 33, 34},
 	}
-
 	framework.
 		NewTest(t).
+		Features("traffic.routing").
```

Comment:
Do you have a link to the currently defined features? Would it make sense to add VMs as a feature?

Reply:
The labels could be found [here](https://github.com/istio/istio/blob/master/pkg/test/framework/features/features.yaml), and according to the README, the label could have a [scenario](https://github.com/istio/istio/blob/master/pkg/test/framework/features/README.md#adding-new-feature-constants). Since the functionality we are testing overlaps (which is also why we are only extending and not adding new tests), how about label it `Features("traffic.routing.vm")`? The `vm` part is the scenario in the test case.

Reply:
Lets follow up with adding VM related coverage as a feature. Can you open an issue and start the discussion?